Updates VMware documentation links and fixes typos

venilnoronha opened the pull request:
This PR adds a redirection for the VMware vSphere getting started guide to VMware's external documentation [website](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/) and updates related hyperlinks. Also, typos at several locations have been fixed.

This PR produces the following changes:
- Adds `301` redirection for `/docs/getting-started-guides/vsphere/` to `https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/`
- Updates hyperlinks from `/docs/getting-started-guides/vsphere/` to `https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/`
- Fixes typos i.e. `VMWare` and `Vmware` -> `VMware`

venilnoronha commented:
/cc @abrarshivani @prashima @rohitjogvmw

abrarshivani commented on docs/setup/pick-right-solution.md:
```diff
@@ -115,9 +115,9 @@ These solutions are combinations of cloud providers and operating systems not co
-* [Vmware vSphere](/docs/getting-started-guides/vsphere/)  (uses Debian)
-* [Vmware vSphere, OpenStack, or Bare Metal](/docs/getting-started-guides/ubuntu/) (uses Juju, Ubuntu and flannel)
-* [Vmware](/docs/getting-started-guides/coreos/)  (uses CoreOS and flannel)
+* [VMware vSphere](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/) (uses Debian)
```
I think it is better to remove ```uses Debian```

venilnoronha replied:
Addressed in 67b461f.

abrarshivani commented on docs/setup/pick-right-solution.md:
```diff
@@ -164,14 +164,14 @@ GCE                  | CoreOS       | CoreOS | flannel     | [docs](/docs/gettin
 Bare-metal (Offline) | CoreOS       | CoreOS | flannel     | [docs](/docs/getting-started-guides/coreos/bare_metal_offline/)              |  Community ([@jeffbean](https://github.com/jeffbean))
 CloudStack           | Ansible      | CoreOS | flannel     | [docs](/docs/getting-started-guides/cloudstack/)                             |  Community ([@sebgoa](https://github.com/sebgoa))
-Vmware vSphere       | Saltstack    | Debian | OVS         | [docs](/docs/getting-started-guides/vsphere/)                                |  Community ([@imkin](https://github.com/imkin))
+VMware vSphere       | Saltstack    | Debian | OVS         | [docs](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/)  |  Community ([@imkin](https://github.com/imkin))
```
Let's update this to latest for other fields too

venilnoronha replied:
Addressed in 67b461f.

prashima commented:
/lgtm

abrarshivani reviewed (approved):
LGTM

venilnoronha committed:
Updates VMware documentation

* Adds 301 redirection for /docs/getting-started-guides/vsphere/ to
  https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/

* Updates hyperlinks from /docs/getting-started-guides/vsphere/ to
  https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/

* Updates solution table entries for VMware vSphere

* Fixes typos i.e. VMWare/Vmware -> VMware
